### PR TITLE
[APB 3626][MP] Pull email through to one more step page

### PR DESCRIPTION
**Pull request by mariannepearson0**

### CODE REVIEW

File: app/uk/gov/hmrc/agentinvitationsfrontend/journeys/AgentInvitationFastTrackJourneyModel.scala [hunk 1]
```diff
@@ -156,7 +158,8 @@ object AgentInvitationFastTrackJourneyModel extends JourneyModel {
       hasPendingInvitationsFor: HasPendingInvitations,
       hasActiveRelationshipFor: HasActiveRelationship,
       createInvitation: CreateInvitation,
-      getAgentLink: GetAgentLink): Future[State] =
+      getAgentLink: GetAgentLink,
+      getAgencyEmail: Future[String]): Future[State] =
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentinvitationsfrontend/journeys/AgentInvitationFastTrackJourneyModel.scala]: May as well have a type alias for this one too, to be consistent with the functions

Comment (johnrevill) [inline app/uk/gov/hmrc/agentinvitationsfrontend/journeys/AgentInvitationFastTrackJourneyModel.scala]: It looks like the real `getAgencyEmail` function has already been called and the result of that function is then passed to this one. 
i.e. should the signature here be `getAgencyEmail: () => Future[String]` instead?

File: app/uk/gov/hmrc/agentinvitationsfrontend/journeys/AgentInvitationFastTrackJourneyModel.scala [hunk 2]
```diff
@@ -240,16 +244,17 @@ object AgentInvitationFastTrackJourneyModel extends JourneyModel {
 
     def checkedDetailsAllInformation(checkPostcodeMatches: CheckPostcodeMatches)(checkDobMatches: CheckDOBMatches)(
       checkRegDateMatches: CheckRegDateMatches)(createInvitation: CreateInvitation)(getAgentLink: GetAgentLink)(
-      hasPendingInvitations: HasPendingInvitations)(hasActiveRelationship: HasActiveRelationship)(
-      featureFlags: FeatureFlags)(agent: AuthorisedAgent)(confirmation: Confirmation) = Transition {
+      getAgencyEmail: Future[String])(hasPendingInvitations: HasPendingInvitations)(
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentinvitationsfrontend/journeys/AgentInvitationFastTrackJourneyModel.scala]: Same thing with the `getAgencyEmail` here and on other methods

File: app/uk/gov/hmrc/agentinvitationsfrontend/journeys/AgentInvitationFastTrackJourneyModel.scala (hunk 1 above)
Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/journeys/AgentInvitationFastTrackJourneyModel.scala]: 👍

File: app/uk/gov/hmrc/agentinvitationsfrontend/journeys/AgentInvitationFastTrackJourneyModel.scala (hunk 2 above)
Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/journeys/AgentInvitationFastTrackJourneyModel.scala]: 👍

**Pull request merged**